tests/integration/test_provider_openai.py

```diff
@@ -0,0 +1,89 @@
+
+api_manager = ApiManager()
+
+
+@pytest.fixture(autouse=True)
+def reset_api_manager():
+    api_manager.reset()
+    yield
+
+
+@pytest.fixture(autouse=True)
+def mock_costs():
+    with patch.dict(
+        COSTS,
+        {
+            "gpt-3.5-turbo": {"prompt": 0.002, "completion": 0.002},
+            "text-embedding-ada-002": {"prompt": 0.0004, "completion": 0},
+        },
+        clear=True,
+    ):
+        yield
+
+
+class TestProviderOpenAI:
+    @staticmethod
+    def test_create_chat_completion_debug_mode(caplog):
+        """Test if debug mode logs response."""
+        messages = [
+            {"role": "system", "content": "You are a helpful assistant."},
+            {"role": "user", "content": "Who won the world series in 2020?"},
+        ]
+        model = "gpt-3.5-turbo"
+        with patch("openai.ChatCompletion.create") as mock_create:
+            mock_response = MagicMock()
+            del mock_response.error
+            mock_response.usage.prompt_tokens = 10
+            mock_response.usage.completion_tokens = 20
+            mock_create.return_value = mock_response
+
+            openai.create_chat_completion(messages, model=model)
+
+            assert "Response" in caplog.text
+
+    @staticmethod
+    def test_create_chat_completion_empty_messages():
+        """Test if empty messages result in zero tokens and cost."""
+        messages = []
+        model = "gpt-3.5-turbo"
+
+        with patch("openai.ChatCompletion.create") as mock_create:
+            mock_response = MagicMock()
+            del mock_response.error
+            mock_response.usage.prompt_tokens = 0
+            mock_response.usage.completion_tokens = 0
+            mock_create.return_value = mock_response
+
+            openai.create_chat_completion(messages, model=model)
+
+            assert api_manager.get_total_prompt_tokens() == 0
+            assert api_manager.get_total_completion_tokens() == 0
+            assert api_manager.get_total_cost() == 0
+
+    @staticmethod
+    def test_create_chat_completion_valid_inputs():
+        """Test if valid inputs result in correct tokens and cost."""
+        messages = [
+            {"role": "system", "content": "You are a helpful assistant."},
+            {"role": "user", "content": "Who won the world series in 2020?"},
+        ]
+        model = "gpt-3.5-turbo"
+
+        with patch("openai.ChatCompletion.create") as mock_create:
+            mock_response = MagicMock()
+            del mock_response.error
+            mock_response.usage.prompt_tokens = 10
+            mock_response.usage.completion_tokens = 20
+            mock_create.return_value = mock_response
+
+            openai.create_chat_completion(messages, model=model)
+
+            assert api_manager.get_total_prompt_tokens() == 10
+            assert api_manager.get_total_completion_tokens() == 20
+            assert api_manager.get_total_cost() == (10 * 0.002 + 20 * 0.002) / 1000
```

Comment:
@collijk Can you point me in the right direction on this on?

Reply:
I'd just ditch it then.  It's not a super important test.  And it's no longer a unit test.